apps/examples/testcase/ta_tc/wifi_manager/itc/itc_wifi_manager_main.c

```diff
@@ -0,0 +1,358 @@
+    printf("AP config: %s(%d), %s(%d), %d %d\n", config.ssid, config.ssid_length, config.passphrase, \
+            config.passphrase_length, config.ap_auth_type, config.ap_crypto_type);
+
+    ret = wifi_manager_connect_ap(&config);
+    TC_ASSERT_EQ("wifi_manager_connect_ap", ret, WIFI_MANAGER_SUCCESS);
+    WIFITEST_WAIT;
+    sleep(5);
+    ret = wifi_utils_disconnect_ap();
+    TC_ASSERT_EQ("wifi_manager_disconnect_ap", ret, WIFI_MANAGER_SUCCESS);
+
+    ret = wifi_manager_deinit();
+    TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_SUCCESS);
+
+    TC_SUCCESS_RESULT();
+}
+
+/**
+ * @testcase         itc_wifi_manager_scan_ap_p
+ * @brief            Initialize  and de-initalization of wifi manager
+ * @scenario
+ * @apicovered
+ * @precondition     wifi_manager_init
+ * @postcondition    wifi_manager_deinit
+ */
+
+static void itc_wifi_manager_scan_ap_p(void)
+{
+    wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+    ret = wifi_manager_init(&wifi_callbacks);
+    TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_SUCCESS);
+
+    ret = wifi_manager_scan_ap();
+    TC_ASSERT_EQ("wifi_manager_scan_ap", ret, WIFI_MANAGER_SUCCESS);
+    sleep(5);
+    ret = wifi_manager_deinit();
+    TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_SUCCESS);
+
+    TC_SUCCESS_RESULT();
+}
+
+/*******************************************************************************************************************/
+
+int wifi_manager_itc(int argc, FAR char *argv[])
+{
+    sem_wait(&tc_sem);
+    working_tc++;
+
+    total_pass = 0;
+    total_fail = 0;
+
+    printf("=== TINYARA WIFI_MANAGER ITC START! ===\n");
+
+    itc_wifi_manager_init_deinit_p();
+
+    itc_wifi_manager_set_mode_p();
+
+    itc_wifi_manager_get_mode_p();	// set softap mode inside this function
+
+    itc_wifi_manager_connect_disconnect_ap_p();	// change to station mode and try to connect to ap
+
+    itc_wifi_manager_scan_ap_p(); // Reinitialized wifi manager with the callback hander for scan results
+
+    printf("\n=== TINYARA WIFI_MANAGER ITC COMPLETE ===\n");
+    printf("\t\tTotal pass : %d\n\t\tTotal fail : %d\n", total_pass, total_fail);
+
+    working_tc--;
+    sem_post(&tc_sem);
+
+    return 0;
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int itc_wifi_manager_main(int argc, char *argv[])
+#endif
+{
+    int ssid_len = strlen(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_SSID);
+    int passphrase_len = strlen(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_PASSPHRASE);
+    int softap_ssid_len =	strlen(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_SOFTAP_SSID);
+    int softap_passphrase_len = strlen(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_SOFTAP_PASSPHRASE);
+
+    if ((ssid_len > 31) || (softap_ssid_len > 31) || (passphrase_len > 63) || (softap_passphrase_len > 63)) {
+        printf("AP or SoftAP configuration fails: too long ssid or passphrase\n");
+        printf("Make sure that length of SSID < 32 and length of passphrase < 64\n");

```

Comment:
How about defining valid macros for 32 and 64 .
Like below
 ` #define SSID_MAX_LEN 32`
 `#define PASSPHRASE_MAX_LEN 64`